requirements.txt

```diff
@@ -1,5 +1,5 @@
 numpy>=1.19.2
-pyspark>=2.4.0
+pyspark>=3.1.1
```

Comment:
is tehre a reason we changed this?  ideally you want your package to be compatible with any spark

Reply:
I changed this because we started importing Spark high-level function in the code e.g. filter otherwise we get an error that filter does not exist in sq.functions

Reply:
still if you think it's better with 2.4 feel free to change it back